src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java

```diff
@@ -54,30 +56,36 @@ private static AmazonApiGateway getAmazonApiGateway() {
     /**
      * Request an API key from AWS api gateway and assign it to an existing usage plan.
      */
-    public static ApiKey createApiKey(String userId, String usagePlanId) {
-        if (userId == null || usagePlanId == null) {
+    public static ApiKey createApiKey(ApiUser user, String usagePlanId) {
+        if (user == null || user.id == null || usagePlanId == null) {
             LOG.error("All required input parameters must be provided.");
             return null;
         }
         long startTime = System.currentTimeMillis();
         try {
             AmazonApiGateway gateway = getAmazonApiGateway();
-            // create API key
+            // Before creating key, verify usage plan exists.
```

Comment:
Perhaps add a comment explaining what happens if the usage plan does not exist. (does that trigger an exception?)

Reply:
Yes, exactly. I added some changes to clarify the behavior (there's a new exception class as well).